templates/admin/config_settings.tmpl

```diff
@@ -39,4 +39,19 @@
 		</div>
 	</form>
 </div>
+
+<h4 class="ui top attached header">
+	{{ctx.Locale.Tr "admin.config.repository_snippets"}}
+</h4>
+<div class="ui attached segment">
+	<form class="ui form form-fetch-action" method="post" action="{{AppSubUrl}}/-/admin/config?key={{.SystemConfig.Repository.SnippetRemoteName.DynKey}}">
+		<div class="field" data-field-patched="true">
+			<label for="remote_name">{{ctx.Locale.Tr "admin.config.repository_snippets.remote_name"}}</label>
+			<input id="remote_name" name="value" value="{{.SystemConfig.Repository.SnippetRemoteName.Value ctx}}" maxlength="100" dir="auto" placeholder="{{.SystemConfig.Repository.SnippetRemoteName.Def}}">
```

Comment:
I don't think the `Def` here is useful.

The value is always filled by `Value` when a user opens this page.

So I think we should remove the `Def` function.

Reply:
The main reason for `Def` is being able to get the default value when user removes value on settings page. 
It is used here: https://github.com/go-gitea/gitea/pull/35172/files#diff-b3b66aef8cc83304cd5cfc661233e6c1c1be656fbd6cd1e09b114348f224ff32R251

In current architecture `func (value *Value[T]) Value(ctx context.Context) (v T)` function uses a default value only if it is not present in the database. So `null` in db, `nil` in go values are valid ones, and therefore the default value won't be used, and as a result there won't be any remote name on the page in snippets.

Reply:
> [](https://github.com/go-gitea/gitea/pull/35172/files#diff-b3b66aef8cc83304cd5cfc661233e6c1c1be656fbd6cd1e09b114348f224ff32R251)
> 
> In current architecture `func (value *Value[T]) Value(ctx context.Context) (v T)` function uses a default value only if it is not present in the database. So `null` in db, `nil` in go values are valid ones, and therefore the default value won't be used, and as a result there won't be any remote name on the page in snippets.

But if the case `null in db` happens, the "Value" itself is already wrong at many places. What would you expect to see in the "git clone command" for this case?

IMO we should fix such edge from root cause, but not add a `Def` method.